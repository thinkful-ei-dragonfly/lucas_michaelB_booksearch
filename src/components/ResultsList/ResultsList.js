import React from 'react'
import ResultItem from '../ResultItem/ResultItem'

function ResultsList(props){
  let bookResults = [
    <div key='0'>No Results.  Use the search bar to find books.</div>
  ];
  if (props.bookResults && props.bookResults.length > 0) {
    bookResults = props.bookResults.map(book => {
      return <ResultItem key={book.id} book={book}/>
    });
  }
    return (
      <ul className="ResultsList">
      {bookResults}
      </ul>
    )
}

export default ResultsList;
