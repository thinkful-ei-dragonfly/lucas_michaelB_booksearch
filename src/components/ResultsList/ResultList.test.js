import React from 'react';
import ReactDOM from 'react-dom';
import ResultsList from './ResultsList';
import '../../setupTests';

import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

const books = [
  {
  id: '1',
  name: 'book 1',
  authors: 'author 1',
  description: 'description 1',
  printType: 'BOOK',
  price: '11.99',
  isEbook: true,
  thumbnail: 'https://learn.getgrav.org/system/images/media/thumb-jpg.png'
  }
]

// it('renders without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<ResultsList bookResults={books}/>, div);
//   ReactDOM.unmountComponentAtNode(div);
// });

describe(`ResultsList Component`, () => {
  it('renders ResultsList without errors no input', () => {
    const wrapper = shallow(<ResultsList />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders ResultsList without errors with sample props', () => {
    const wrapper = shallow(<ResultsList  bookResults={books}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
