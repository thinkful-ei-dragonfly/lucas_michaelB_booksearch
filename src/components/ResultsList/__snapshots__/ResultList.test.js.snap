// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsList Component renders ResultsList without errors no input 1`] = `
<ul
  className="ResultsList"
>
  <div
    key="0"
  >
    No Results.  Use the search bar to find books.
  </div>
</ul>
`;

exports[`ResultsList Component renders ResultsList without errors with sample props 1`] = `
<ul
  className="ResultsList"
>
  <ResultItem
    book={
      Object {
        "authors": "author 1",
        "description": "description 1",
        "id": "1",
        "isEbook": true,
        "name": "book 1",
        "price": "11.99",
        "printType": "BOOK",
        "thumbnail": "https://learn.getgrav.org/system/images/media/thumb-jpg.png",
      }
    }
    key="1"
  />
</ul>
`;
